File: Algorithms/src/nattybear/level1/Budget.js
function solution(d, budget) {

  let sum = 0;
  let acc = 0;

  d.sort((x, y) => x - y);

  for (let e of d) {
    sum += e;
    if (sum <= budget)
      acc++;
    else
      break;
  }

  return acc;
}

console.log(solution([1, 3, 2, 5, 4], 9));
console.log(solution([2, 2, 3, 3], 10));

File: Algorithms/src/nattybear/level1/Budget.java
package nattybear.level1;

import java.util.Arrays;

public class Budget {
	private static class Solution {
		public int solution(int[] d, int budget) {

			int sum = 0;
			int num = 0;

			Arrays.sort(d);

			for (int e : d) {
				sum += e;
				if (sum <= budget) {
					num++;
				} else {
					break;
				}
			}

			return num;
		}
	}

	public static void main(String[] args) {
		Solution s = new Solution();
		System.out.println(s.solution(new int[] { 1, 3, 2, 5, 4 }, 9));
		System.out.println(s.solution(new int[] { 2, 2, 3, 3 }, 10));
	}
}
